lib/components/viewers/nearby/nearby-view.tsx

```diff
@@ -0,0 +1,183 @@
+import Loading from '../../narrative/loading'
+import MobileContainer from '../../mobile/container'
+import MobileNavigationBar from '../../mobile/navigation-bar'
+
+import {
+  FloatingLoadingIndicator,
+  NearbySidebarContainer,
+  Scrollable
+} from './styled'
+import RentalStation from './rental-station'
+import Stop from './stop'
+import Vehicle from './vehicle-rent'
+import VehicleParking from './vehicle-parking'
+
+const AUTO_REFRESH_INTERVAL = 15000
+
+type LatLonObj = { lat: number; lon: number }
+
+type Props = {
+  fetchNearby: (latLon: LatLonObj, map?: MapRef) => void
+  getCurrentPosition?: (
+    intl: IntlShape,
+    setAsType?: string | null,
+    onSuccess?: (position: GeolocationPosition) => void
+  ) => void
+  hideBackButton?: boolean
+  mobile?: boolean
+  nearby: any
+  nearbyViewCoords?: LatLonObj
+  setHighlightedLocation: (location: Location | null) => void
+  setMainPanelContent: (content: number) => void
+  viewNearby?: (pos: LatLonObj) => void
+}
+
+const getNearbyItem = (place: any) => {
+  switch (place.__typename) {
+    case 'RentalVehicle':
+      return <Vehicle key={place.id} vehicle={place} />
+    case 'Stop':
+      return <Stop showOperatorLogo stopData={place} />
+    case 'VehicleParking':
+      return <VehicleParking place={place} />
+    case 'BikeRentalStation':
+      return <RentalStation place={place} />
+    default:
+      console.warn(
+        `Received unsupported nearby place type: ${place.__typename} `
+      )
+      return null
+  }
+}
+
+function NearbyView(props: Props): JSX.Element {
+  const {
+    fetchNearby,
+    getCurrentPosition: getPosition,
+    mobile,
+    nearby,
+    nearbyViewCoords,
+    setHighlightedLocation,
+    setMainPanelContent,
+    viewNearby
+  } = props
+  const map = useMap().current
+  const intl = useIntl()
+  const [loading, setLoading] = useState(true)
+  const firstItemRef = useRef<HTMLDivElement>(null)
+  useEffect(() => {
+    firstItemRef.current?.scrollIntoView({ behavior: 'smooth' })
+    if (nearbyViewCoords) {
+      fetchNearby(nearbyViewCoords, map)
+      setLoading(true)
+      const interval = setInterval(() => {
+        fetchNearby(nearbyViewCoords, map)
+        setLoading(true)
+      }, AUTO_REFRESH_INTERVAL)
+      return function cleanup() {
+        clearInterval(interval)
+      }
+    } else {
+      if (getPosition && viewNearby) {
+        getPosition(intl, null, (pos) => {
+          viewNearby({ lat: pos.coords.latitude, lon: pos.coords.longitude })
+        })
+      }
+    }
+  }, [nearbyViewCoords])
+
+  const onMouseEnter = useCallback(
+    (location: Location) => {
+      setHighlightedLocation(location)
+    },
+    [setHighlightedLocation]
+  )
+  const onMouseLeave = useCallback(() => {
+    setHighlightedLocation(null)
+  }, [setHighlightedLocation])
+
+  const nearbyItemList = nearby?.map((n: any) => (
+    <div
+      key={n.place.id}
+      onMouseEnter={() => onMouseEnter(n.place)}
```

Comment:
Add a similar behavior for `onFocus` and `onBlur` (otherwise, it is impossible to highlight the stop using the keyboard alone).